src/units/attack_type.hpp

```diff
@@ -97,6 +97,8 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	unit_ability_list get_weapon_ability(const std::string& ability) const;
 	/** Returns list who contains get_weapon_ability and get_specials list for each ability type */
 	unit_ability_list get_specials_and_abilities(const std::string& special) const;
+	/** Returns list who contains get_specials_and_abilities with highest priority specified by value of attribute 'type_priority' for each ability plague or damage_type*/
+	unit_ability_list get_specials_and_abilities_with_string(const std::string& special) const;
```

Comment:
This is a terrible name for this function. "with_string" gives no indication whatsoever about how it differs from `get_specials_and_abilities`, since both take a single string argument with the same name.

Doesn't "with_priority" make more sense?

Reply:
corrected